test/integration/deployment/deployment_test.go

```diff
@@ -1175,10 +1186,16 @@ func TestReplicaSetOrphaningAndAdoptionWhenLabelsChange(t *testing.T) {
 	}
 
 	// Wait for the deployment to create a new replicaset
+	// This will trigger collision avoidance due to deterministic nature of replicaset name
+	// i.e., the new replicaset will have a name with different hash to preserve name uniqueness
```

Comment:
Collision avoidance will also modify selectors, but this comment should be about different replicaset name, so I didn't mention about the selector change here. Perhaps we can include a link to the relevant design doc section.